zebra-consensus/src/transaction.rs

```diff
@@ -403,4 +444,58 @@ where
             )),
         }
     }
+
+    /// Verifies if a transaction's transparent `inputs` are valid using the provided
+    /// `script_verifier`.
+    fn verify_transparent_inputs_and_outputs(
+        request: &Request,
+        network: Network,
+        inputs: &[transparent::Input],
+        script_verifier: script::Verifier<ZS>,
+    ) -> Result<AsyncChecks, TransactionError> {
+        let transaction = request.transaction();
+
+        if transaction.is_coinbase() {
+            check::coinbase_tx_no_prevout_joinsplit_spend(&transaction)?;
+
+            Ok(FuturesUnordered::new())
+        } else {
+            // feed all of the inputs to the script and shielded verifiers
+            // the script_verifier also checks transparent sighashes, using its own implementation
+            let cached_ffi_transaction = Arc::new(CachedFfiTransaction::new(transaction));
+            let known_utxos = request.known_utxos();
+            let upgrade = request.upgrade(network);
+
+            let script_checks = (0..inputs.len())
+                .into_iter()
+                .map(move |input_index| {
+                    let request = script::Request {
+                        upgrade,
+                        known_utxos: known_utxos.clone(),
+                        cached_ffi_transaction: cached_ffi_transaction.clone(),
+                        input_index,
+                    };
+
+                    script_verifier.clone().oneshot(request).boxed()
+                })
+                .collect();
+
+            Ok(script_checks)
+        }
+    }
```

Comment:
We may want to move this struct function into the `transaction::checks` module

Reply:
If we leave the functions on the transaction verifier, they will be more discoverable.

But I don't think it makes a big difference either way, because they are all called directly from the verifier code.

Reply:
I'm not sure how to proceed here :confused:  I left it as is, maybe we should revisit this discussion once the refactor of the other methods is complete?